### PR TITLE
Remove object from Timeout types.

### PR DESCRIPTION
**Pull request by hramezani**
Closes https://github.com/urllib3/urllib3/issues/2252
Thanks, @sethmlarson for the solution.

### CODE REVIEW

File: src/urllib3/connectionpool.py [hunk 1]
```diff
@@ -56,10 +56,10 @@
 
 log = logging.getLogger(__name__)
 
-_Default = object()
+_Default: float = cast(float, object())
 
 
-_TYPE_TIMEOUT = Union[Timeout, int, float, object]
+_TYPE_TIMEOUT = Union[Timeout, int, float]
```
Comment (bluetech) [inline src/urllib3/connectionpool.py:62]: Unrelated to the PR: in python typing (unlike in runtime), `int` is actually a sub-type of `float`, so the `int` here is redundant. However it makes to keep for clarity.

File: src/urllib3/connectionpool.py [hunk 2]
```diff
@@ -56,10 +56,10 @@
 
 log = logging.getLogger(__name__)
 
-_Default = object()
+_Default: float = cast(float, object())
```
Comment (bluetech) [inline src/urllib3/connectionpool.py:59]: Some other options:

- Mark the sentinels as `Any`, then you don't need the cast, but the default parameter assignment still works. And people would perhaps be less tempted to use them as actual floats.

- Add a real sentinel type (e.g. a one-item enum) and add it to the types and all. This is completely type safe but somewhat annoying.

I think all approaches are probably fine.

**Review by bluetech — APPROVED** [review]

File: src/urllib3/connectionpool.py (hunk 2 above)
Comment (hramezani) [reply src/urllib3/connectionpool.py:59]: Thanks, @bluetech for the review and suggestion.
Yeah, you are right. we have to add a real sentinel type (based on the discussion in the dev channel)

**Comment by hramezani** [comment]
Closing per @bluetech suggestion and discussion with other maintainers)

**Pull request closed without merging**